lib/cosmos/interfaces/protocols/protocol.rb

```diff
@@ -15,9 +15,12 @@
 module Cosmos
   class Protocol
     attr_accessor :interface
+    attr_accessor :allow_empty_data
 
-    def initialize
+    # @param allow_empty_data [true/false] Whether STOP should be returned on empty data
+    def initialize(allow_empty_data = false)
```

Comment:
Actually this needs to be false with overrideprotocol by itself.  Anyways, there are some cases where you might want to allow empty packets to get through (to log blank lines for example) and this supports that.